### PR TITLE
[Documentation] 05-Pipeline Launch options

### PR DESCRIPTION
**Pull request by MounaSafiHarab**

### CODE REVIEW

File: docs/05-PipelineOptions.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+imaging specialist launching the pipeline (manually or through a cron job) at pre-defined 
+time intervals (e.g. once a day, or once a week).
+
+```
+uploadNeuroDB/imaging_upload_file_cronjob.pl -profile prod -verbose
+```
+
+
+## 5.3 Option 3
+This is an option that addresses restrospecively collected data where uploading hundreds
```
Comment (cmadjar) [inline docs/05-PipelineOptions.md]: typo in retrospectively

File: docs/05-PipelineOptions.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+/data/incoming/Lego_Phantom_MNI_20140101.zip Y
+```
+
+The insertion pipeline can then be triggered using the command:
+```
+./batch_uploads_imageuploader -profile prod < scans_list.txt >log_batch_imageuploader.txt 2>&1 
+
+```
+
+## 5.4 Option 4 (why exactly would one want to do this?)
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/05-PipelineOptions.md]: I really wonder who would want to do such a thing ;)

**Comment by MounaSafiHarab** [comment]
@cmadjar 

ALL valuable comments. I will address as soon as I get some free minutes. Thanks for going through it all!

File: docs/05-PipelineOptions.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/05-PipelineOptions.md]: Actually, while going through the wiki documentation for the install, I noticed this section:
https://github.com/aces/Loris/wiki/Imaging-Database#multi-step-pipeline-execution

This is linked when we want to rerun the Imaging pipeline with the trouble shooting. Can we talk about this on Monday? Might be easier than via Github comments.

File: docs/05-PipelineOptions.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+In the next three sub-sections, the three launch options illusrtated in Figure 5.1 
+will be briefly highlighted and the exact command needed to launch the pipeline shown.
+Details about the scripts themselves can be found in the Scripts section.
+ 
+## 5.1 Option 1
+
+This is a typical option for a project prospectively collecting data with a) multiple
+sites involved, b) a designated user per site for collecting scans from the scanner 
+console and uploading to LORIS, and c) a project imaging specialist monitoring and
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/05-PipelineOptions.md]: no, there are possibilities to have imaging uploader send automated emails as soon as an upload happens, but not automated (this is why I say someone is monitoring). I reworded this sentence a bit...

**Comment by MounaSafiHarab** [comment]
@cmadjar 

changes incorporated from your comments here and one of the meeting we had (namely the LORIS BLUE and grey etc...!)
I also took a couple of lines from the wiki...

let me know if this version is better

File: docs/05-PipelineLaunchOptions.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# 5.0 - Pipeline Launch options
+
+Scans upload into LORIS and insertion pipeline's triggering can be done in a few 
+ways depending on each project's needs. The best choice will depend on whether 
+for example data is collected retrospectively or prospectively, and/or whether 
+the MRI protocol is harmonized across all the sites involved in the study.
+
+The different options available are graphically illustrated in the graph below.
+
+![PipelineLaunchOptions](images/PipelineLaunchOptions.png)
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: Love the graph!!!

See picture below for a tiny comment on the graph
pipelinelaunchoptions

File: docs/05-PipelineLaunchOptions.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
+# 5.0 - Pipeline Launch options
+
+Scans upload into LORIS and insertion pipeline's triggering can be done in a few 
+ways depending on each project's needs. The best choice will depend on whether 
+for example data is collected retrospectively or prospectively, and/or whether 
+the MRI protocol is harmonized across all the sites involved in the study.
+
+The different options available are graphically illustrated in the graph below.
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: Remove "graphically".

File: docs/05-PipelineLaunchOptions.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# 5.0 - Pipeline Launch options
+
+Scans upload into LORIS and insertion pipeline's triggering can be done in a few 
+ways depending on each project's needs. The best choice will depend on whether 
+for example data is collected retrospectively or prospectively, and/or whether
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: I would remove the "for example". Makes the sentence more complicated to read.

Or could rephrase like this?
"The best choice will depend on the study workflow and protocol. For example, it will depend on whether data is collected retrospectively or prospectively, ..."

File: docs/05-PipelineLaunchOptions.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+Details about the scripts themselves can be found in the [Scripts](04-Scripts.md)
+section.
+ 
+## 5.1 Option 1
+
+Triggering the pipeline is done from the `/data/$PROJECT/bin/mri` directory as 
+follows:
+
+```
+uploadNeuroDB/imaging_upload_file.pl -profile prod -upload_id $UploadIDID /PATH/TO/UPLOAD/PSCID_CandID_VisitLabel_OptionalSuffix.zip -verbose
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: I don't think you require "uploadNeuroDB/" when calling the script since it is supposed to be in the $PATH variable. Technically, you could trigger imaging_upload_file.pl from anywhere, no?

File: docs/05-PipelineLaunchOptions.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+3. a project's imaging specialist monitoring new uploads, and launching the 
+insertion pipeline manually, for every new `UploadID` separately. 
+
+## 5.2 Option 2 
+
+Triggering the pipeline from the `/data/$PROJECT/bin/mri` directory can also be 
+achieved as follows: 
+
+```
+uploadNeuroDB/imaging_upload_file_cronjob.pl -profile prod -verbose
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: same comment here, don't think you need to specify "uploadNeuroDB/"

File: docs/05-PipelineLaunchOptions.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+uploads (the script can be added to the system's crontab to launch at 
+pre-scheduled times, as per the project's requirements).
+
+
+
+## 5.3 Option 3
+
+The insertion pipeline can then be triggered using the command:
+```
+./batch_uploads_imageuploader -profile prod < scans_list.txt >log_batch_imageuploader.txt 2>&1
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: don't need the "./" to run "batch_uploads_imageuploader"

File: docs/05-PipelineLaunchOptions.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+This is an option that addresses retrospectively collected data where uploading 
+hundreds of scans, one scan at a time, using the LORIS Imaging Uploader a 
+user-friendly interface is impractical. It is also the option of choice for 
+prospective studies that want to benefit from tracking scans through the Imaging 
+Uploader while automating the upload and insertion process without a user/GUI 
+interface interaction. 
+
+In this case, the scans should be transferred to the LORIS-MRI filesystem (a 
+path such as `/data/incoming/`), and on input text file `scans_list.txt`
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: would replace "a path such as " by "for example in "

Comment (cmadjar) [reply docs/05-PipelineLaunchOptions.md]: also, "on" is probably meant to be "**one** input text file"?

Comment (cmadjar) [reply docs/05-PipelineLaunchOptions.md]: would cut the sentence in two:
"In this case, the scans should be transferred to the LORIS-MRI filesystem (for example in `/data/incoming/`). Additionally, an input text file (such as `scans_list.txt`) listing one dataset per line should be provided (see example below)."

File: docs/05-PipelineLaunchOptions.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+This is an option that addresses retrospectively collected data where uploading 
+hundreds of scans, one scan at a time, using the LORIS Imaging Uploader a 
+user-friendly interface is impractical. It is also the option of choice for 
+prospective studies that want to benefit from tracking scans through the Imaging 
+Uploader while automating the upload and insertion process without a user/GUI 
+interface interaction. 
+
+In this case, the scans should be transferred to the LORIS-MRI filesystem (a 
+path such as `/data/incoming/`), and on input text file `scans_list.txt` 
+provided, listing one line pr dataset (example below).
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: typo, should be "**per** dataset" instead of "**pr** dataset"

File: docs/05-PipelineLaunchOptions.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+The following fields are expected, separated by spaces:
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: would rephrase this to:

"Each line in that text file should include the following information separated by spaces:"

File: docs/05-PipelineLaunchOptions.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+1. the full path to the zipped DICOM scanset (`.zip`, `.tgz`, `.tar.gz`),
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: replace scanset by dataset?

File: docs/05-PipelineLaunchOptions.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+interface interaction. 
+
+In this case, the scans should be transferred to the LORIS-MRI filesystem (a 
+path such as `/data/incoming/`), and on input text file `scans_list.txt` 
+provided, listing one line pr dataset (example below).
+The following fields are expected, separated by spaces:
+
+1. the full path to the zipped DICOM scanset (`.zip`, `.tgz`, `.tar.gz`), 
+2. Y or N depending on whether the scan is for a phantom or not, and
+3. patient name following the PSCID_CandID_VisitLabel LORIS convention for real
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: add 'the' in front of 'patient name' (like in 1.)

File: docs/05-PipelineLaunchOptions.md (hunk 4 above)
Comment (MounaSafiHarab) [reply docs/05-PipelineLaunchOptions.md]: yes, I did that in purpose to draw attention that this is useful in case of multiple uploads, but I can remove it (especially if it was not obvious to you why I did it), not a problem!

File: docs/05-PipelineLaunchOptions.md (hunk 7 above)
Comment (MounaSafiHarab) [reply docs/05-PipelineLaunchOptions.md]: you could, BUT, 1) our scripts internally are called/run this way (they all call the bin/mri/ directory and append the uploadNeuroDB/ to the path), and 2) users that want to do auto-complete (like I do), have to put the leading path...

File: docs/05-PipelineLaunchOptions.md (hunk 9 above)
Comment (MounaSafiHarab) [reply docs/05-PipelineLaunchOptions.md]: same for the auto-complete comment above... probably here due to the lack of extension!

**Comment by MounaSafiHarab** [comment]
I think I took care of your feedback (except the uploadNeuroDB/ part; please see my comment). I do not mind changing it if you insist!

the rest of the text, I changed it as requested ! (I took some from the wiki, but I did change it anyway based on your feedback).

thanks!

File: docs/05-PipelineLaunchOptions.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+launches it. As such, this option is used in similar scenarios as Option 1, 
+but with the added advantage of not needing to continuously monitor for new 
+uploads (the script can be added to the system's crontab to launch at 
+pre-scheduled times, as per the project's requirements).
+
+
+
+## 5.3 Option 3
+
+The insertion pipeline can then be triggered using the command:
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: replace "then" by "also"

File: docs/05-PipelineLaunchOptions.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+Triggering the pipeline from the `/data/$PROJECT/bin/mri` directory can also be 
+achieved as follows: 
+
+```
+uploadNeuroDB/imaging_upload_file_cronjob.pl -profile prod -verbose
+```
+                                
+The script `imaging_upload_file_cronjob.pl` does not require a specific 
+`$UploadID` as an argument because it automatically cycles through 
+I<all new uploads> on which the pipeline has not been triggered yet, and
```
Comment (cmadjar) [inline docs/05-PipelineLaunchOptions.md]: "all new uploads" does not display in the MD file. It just displays "through I on".

File: docs/05-PipelineLaunchOptions.md (hunk 15 above)
Comment (MounaSafiHarab) [reply docs/05-PipelineLaunchOptions.md]: good catch!

**Comment by MounaSafiHarab** [comment]
> @MounaSafiHarab 2 minor things to fix.

done!
